Source/WebCore/crypto/algorithms/CryptoAlgorithmECDH.cpp

```diff
@@ -94,7 +94,7 @@ void CryptoAlgorithmECDH::deriveBits(const CryptoAlgorithmParameters& parameters
             exceptionCallback(ExceptionCode::OperationError);
             return;
         }
-        if (!length || !(*length)) {
+        if (!length) {
```

Comment:
This technically does work. But I think it would be wise to check for null as the first thing in the function ::deriveBits and bail early.
For Zero, I think we can check right before this lambda and don't even get into doing scalar multiplication and just return a zero length result.

Reply:
I agree on the optimization for the zero length, which would even allow us to avoid the call to the platformDeriveBits key. 

However I'm not so sure about the early bail in case of null; since the change to define length as optional, null means that we should return the full derivedKey. Hence, I don't think we save much to avoid doing the check inside the lambda.

Reply:
Also, not a strong opinion, but I think it's better to do the length = 0 after the rest of the basic checks (eg. ec parameters, key type, algorithm name). Otherwise we would hide failures in those checks when passing 0 as length.

Reply:
> Otherwise we would hide failures in those checks when passing 0 as length.
It's true vice versa. If we were adding those other checks in this PR instead of zero length, do you think the same argument applies ? I might be missing stuff in the spec. I had a quick look.

Reply:
>  optional, null means that we should return the full derivedKey. 

I did not know that. Then I guess we need to derive the key.

Reply:
> > Otherwise we would hide failures in those checks when passing 0 as length.
> 
> It's true vice versa. If we were adding those other checks in this PR instead of zero length, do you think the same argument applies ? I might be missing stuff in the spec. I had a quick look.

Very true; my preference is based on the fact that the 'length' behavior is based on the specific algorithm ; for instance, PBKDF2 and HKDF would throw in case of 0, and we are aiming for ECDH and X25519 to throw on any truncation. So in my mind it makes more sense to ensure first that the algorithm name, key type and usages are validates and only then check the specific values of the 'length' parameter. 

But as I said, I don't have a strong opinion on this, because you have a valid point and perhaps it's more efficient to early return based on an "cheap" check of the length's value, avoiding the other checks.